// duitku/api/getTopUpInstructionOvo.js
const crypto = require('crypto-js');
const axios = require('axios');
const appRoot = require('app-root-path');
const duitkuConfig = require(`${appRoot}/config/duitku-configuration`);

module.exports = function getTopUpInstructionOvo(credentialCode, channel, cb){
    let timestamp = new Date().getTime();
    let passport = duitkuConfig.passport
    let url = ': https://accountlink-prod.duitku.com/api/accountlinking/v1/topup/intructions';
    if(!passport) url = 'https://api-sandbox.duitku.com/accountlinking/api/accountlinking/v1/topup/intructions';
    let data = {
        credentialCode,
        channel
    }
    
    axios({
        method: 'POST',
        url: url,
        data,
        headers: {
                    "Accept": "application/json",
                    "Content-type" : "application/json; charset=UTF-8",
                    "X-duitku-signature": crypto.SHA256(`${duitkuConfig.merchantCode}${timestamp}${JSON.stringify(data)}${duitkuConfig.apiKey}`).toString(),
                    "X-duitku-timestamp": `${timestamp}`,
                    "X-duitku-merchant": `${duitkuConfig.merchantCode}`,
                }
    })
    .then(response => cb(response.data.data))
    .catch(err =>{
        console.log(err)
        let errorJson = {
            error: err.code,
            status: err.response.status,
            statusMessage: err.response.statusText,
            reason: err.response.data,
        }
        cb({}, errorJson)
    })
}
